### PR TITLE
Use VecGeom's new NavStatePoolView to access the NavStatePool content on device

### PR DESCRIPTION
**Pull request by pcanal**
Requires: https://gitlab.cern.ch/VecGeom/VecGeom/-/merge_requests/803

### CODE REVIEW

File: src/geometry/detail/VGNavCollection.cc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//----------------------------------*-C++-*----------------------------------//
+           //----------------------------------*-C++-*----------------------------------//
```
Comment (sethrj) [inline src/geometry/detail/VGNavCollection.cc:1]: This is a typo

**Review by sethrj — COMMENTED** [review]
Thanks @pcanal , looks good aside from the silly accident at the beginning of file. I guess this supersedes #212? Once the upstream merges are in we can update the CI on this branch and merge.

**Comment by pcanal** [comment]
> I guess this supersedes #212? 

Because of the squash-on-merge, it is more of a (soft) require than supersedes. i.e. We should merge #212 and then this.

File: src/geometry/detail/VGNavCollection.cc (hunk 1 above)
Comment (pcanal) [inline src/geometry/detail/VGNavCollection.cc:1]:
```suggestion
//----------------------------------*-C++-*----------------------------------//
```

**Review by sethrj — APPROVED** [review]

**Pull request merged**